test/TestUtil.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+const assert = require('assert');
+const print = require('../lib/util/print');
+const flatten = require('../lib/util/flattenOperands');
+const math = require('mathjs');
+
+// Test helper contains methods for reducing the redundant code. (Test Util)
+const TestUtil = {};
+
+// Test the distribute steps in the expression string.
+TestUtil.testDistributeSteps = function (evaluator, exprString, outputList) {
```

Comment:
I would get rid of this and just use `testSubsteps` instead, and make `lastString` in the test files before calling it

or at least, do this to reduce duplicated code:

```
TestUtil.testDistributeSteps =  function (evaluator, exprString, outputList) {
  const lastString = outputList[outputList.length - 1];
  TestUtil.testSubsteps (evaluator, exprString, outputList, lastString);
}
``` 

Also, it looks like you use this function for a lot more than distribution. If you want to keep it, definitely rename it